### PR TITLE
trzsz-go 0.1.7 (new formula)

### PR DESCRIPTION
**Pull request by lonnywong**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/trzsz-go.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  url "https://github.com/trzsz/trzsz-go/archive/refs/tags/v0.1.7.tar.gz"
+  sha256 "93fb87141bb8306fbb9a465c4cfe5401dd0646bc1180caf1cd50b2503c93e454"
+  license "MIT"
+
+  depends_on "go" => :build
+  depends_on "make" => :build
+
+  def install
+    system "make"
+    bin.install Dir["bin/*"]
```
[the remark this answers is not in the document]
Comment (lonnywong) [reply Formula/trzsz-go.rb:14]: Errno::ENOENT: No such file or directory - bin

[the remark this answers is not in the document]
Comment (lonnywong) [reply Formula/trzsz-go.rb:14]: How about `prefix.install Dir["bin/*"]` ?

[the remark this answers is not in the document]
Comment (lonnywong) [reply Formula/trzsz-go.rb:14]: I learn `bin.install Dir["output/*"]` from https://docs.brew.sh/Formula-Cookbook

[the remark this answers is not in the document]
Comment (lonnywong) [reply Formula/trzsz-go.rb]: I misunderstood. `go build` has linked the binaries to the homebrew bin path. The `bin.install` is redundant.

[the remark this answers is not in the document]
Comment (lonnywong) [reply Formula/trzsz-go.rb]: done

File: Formula/trzsz-go.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    system "go", "build", "-o", bin/"trzsz", "./cmd/trzsz"
+  end
+
+  test do
+    assert_match "executable file not found", shell_output("#{bin}/trzsz cmd_not_exists 2>&1", 255)
+    touch "tmpfile"
+    assert_match "Not a directory", shell_output("#{bin}/trz tmpfile 2>&1", 254)
+    rm "tmpfile"
+    assert_match "No such file", shell_output("#{bin}/tsz tmpfile 2>&1", 255)
+  end
```
Comment (chenrui333) [inline Formula/trzsz-go.rb:22]:
```suggestion

    assert_match "trz (trzsz) go #{version}", shell_output("#{bin}/trz --version")
    assert_match "tsz (trzsz) go #{version}", shell_output("#{bin}/tsz --version")
    assert_match "trzsz go #{version}", shell_output("#{bin}/trzsz --version")
  end
```

Comment (lonnywong) [reply Formula/trzsz-go.rb:26]: Thanks @chenrui333

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@lonnywong, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**